src/rust/engine/process_execution/src/lib.rs

```diff
@@ -46,6 +47,26 @@ pub mod local;
 pub mod remote;
 pub mod speculate;
 
+extern crate uname;
+
+#[derive(PartialOrd, Ord, Clone, Debug, Eq, PartialEq, Hash)]
```

Comment:
If you derive `Copy` you can get rid of a lot of `.clone()` explicit calls you currently have